suite/contracts/contracts_keeper_test.go

```diff
@@ -0,0 +1,155 @@
+				&contracts.KeeperRegistryOpts{
+					LinkAddr:             s.Link.Address(),
+					ETHFeedAddr:          ef.Address(),
+					GasFeedAddr:          gf.Address(),
+					PaymentPremiumPPB:    uint32(200000000),
+					BlockCountPerTurn:    big.NewInt(3),
+					CheckGasLimit:        checkGasLimit,
+					StalenessSeconds:     big.NewInt(90000),
+					GasCeilingMultiplier: uint16(1),
+					FallbackGasPrice:     big.NewInt(2e11),
+					FallbackLinkPrice:    big.NewInt(2e18),
+				},
+			)
+			Expect(err).ShouldNot(HaveOccurred())
+			err = registry.Fund(s.Wallets.Default(), big.NewFloat(0), big.NewFloat(1))
+			Expect(err).ShouldNot(HaveOccurred())
+			consumer, err = s.Deployer.DeployKeeperConsumer(s.Wallets.Default(), big.NewInt(5))
+			Expect(err).ShouldNot(HaveOccurred())
+			err = consumer.Fund(s.Wallets.Default(), big.NewFloat(0), big.NewFloat(1))
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		By("Registering upkeep target", func() {
+			registrar, err := s.Deployer.DeployUpkeepRegistrationRequests(
+				s.Wallets.Default(),
+				s.Link.Address(),
+				big.NewInt(0),
+			)
+			Expect(err).ShouldNot(HaveOccurred())
+			err = registry.SetRegistrar(s.Wallets.Default(), registrar.Address())
+			Expect(err).ShouldNot(HaveOccurred())
+			err = registrar.SetRegistrarConfig(
+				s.Wallets.Default(),
+				true,
+				uint32(999),
+				uint16(999),
+				registry.Address(),
+				big.NewInt(0),
+			)
+			Expect(err).ShouldNot(HaveOccurred())
+			req, err := registrar.EncodeRegisterRequest(
+				"upkeep_1",
+				[]byte("0x1234"),
+				consumer.Address(),
+				checkGasLimit,
+				s.Wallets.Default().Address(),
+				[]byte("0x"),
+				big.NewInt(9e18),
+				0,
+			)
+			Expect(err).ShouldNot(HaveOccurred())
+			err = s.Link.TransferAndCall(s.Wallets.Default(), registrar.Address(), big.NewInt(9e18), req)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		By("Adding Keepers and a job", func() {
+			keys, err := nodes[0].ReadETHKeys()
+			Expect(err).ShouldNot(HaveOccurred())
+			na := keys.Data[0].Attributes.Address
+			for _, cla := range nodeAddresses {
+				nodeAddressesStr = append(nodeAddressesStr, cla.Hex())
+			}
+			payees := []string{
+				consumer.Address(),
+				consumer.Address(),
+				consumer.Address(),
+				consumer.Address(),
+				consumer.Address(),
+			}
+			err = registry.SetKeepers(s.Wallets.Default(), nodeAddressesStr, payees)
+			Expect(err).ShouldNot(HaveOccurred())
+			_, err = nodes[0].CreateJob(&client.KeeperJobSpec{
+				Name:            "keeper",
+				ContractAddress: registry.Address(),
+				FromAddress:     na,
+			})
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+	})
+	Describe("with Keeper job", func() {
+		It("performs upkeep of a target contract", func() {
+			err = retry.Do(func() error {
+				cnt, err := consumer.Counter(context.Background())
+				if err != nil {
+					return err
+				}
+				if cnt.Int64() == 0 {
+					return errors.New("awaiting for upkeep")
+				}
+				log.Debug().Int64("Counter", cnt.Int64()).Send()
```

Comment:
Add some keys here to what this counter is, as the log line is just:
```
10:21AM DBG  Counter=1
```

Also add some Info logging that outputs when upkeep was performed.